File: src/sp/it/pl/layout/area/ContainerNode.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package sp.it.pl.layout.area;

import javafx.scene.Parent;
import javafx.scene.layout.Pane;
import sp.it.pl.layout.AltState;

public interface ContainerNode extends AltState {

    Pane getRoot();

    default Parent getParent() {
        return getRoot()==null ? null : getRoot().getParent();
    }

    default void close() {}

}

File: src/sp/it/pl/layout/area/ContainerNode.kt
package sp.it.pl.layout.area

import javafx.scene.Parent
import javafx.scene.layout.Pane
import sp.it.pl.layout.AltState

/** Container graphics. */
interface ContainerNode: AltState {

    /** Scene graph root of this object. */
    val root: Pane

    @JvmDefault
    fun getParent(): Parent = root.parent

    @JvmDefault
    fun close() {}

}